modules/currency-l1/src/main/scala/org/tessellation/currency/l1/http/Routes.scala

```diff
@@ -1,29 +1,61 @@
 package org.tessellation.currency.l1.http
 
+import cats.data.Validated.{Invalid, Valid}
 import cats.effect.Async
 import cats.effect.std.{Queue, Supervisor}
 import cats.syntax.all._
 
-import org.tessellation.currency.DataUpdate
+import org.tessellation.currency.l1.domain.error.{InvalidDataUpdate, InvalidSignature}
+import org.tessellation.currency.{BaseDataApplicationL1Service, DataUpdate}
 import org.tessellation.dag.l1.domain.dataApplication.consensus.ConsensusInput
+import org.tessellation.ext.http4s.error._
+import org.tessellation.kryo.KryoSerializer
 import org.tessellation.sdk.domain.cluster.storage.L0ClusterStorage
+import org.tessellation.security.SecurityProvider
 import org.tessellation.security.signature.Signed
 
 import io.circe.Decoder
+import io.circe.shapes._
 import org.http4s.HttpRoutes
-import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.circe.CirceEntityCodec.{circeEntityDecoder, circeEntityEncoder}
 import org.http4s.dsl.Http4sDsl
 import org.typelevel.log4cats.slf4j.Slf4jLogger
+import shapeless._
+import shapeless.syntax.singleton._
 
-final case class Routes[F[_]: Async](
+final case class Routes[F[_]: Async: KryoSerializer: SecurityProvider](
```

Comment:
Imo the name is too generic. How about `DataApplicationRoutes`?

Reply:
fine for me, I'll change that (but I used the same pattern we already have in DAG L1)